diff -rupN orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2020-02-11 22:33:47.000000000 +0100
+++ patched/CMakeLists.txt	2025-02-25 10:29:03.907997882 +0100
@@ -6,7 +6,7 @@ endif()
 #
 PROJECT(libarchive C)
 #
-SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/build/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/build/cmake")
 if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
   set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${libarchive_BINARY_DIR}/bin)
 endif()
@@ -445,6 +445,7 @@ IF(ZLIB_FOUND)
 ENDIF(ZLIB_FOUND)
 MARK_AS_ADVANCED(CLEAR ZLIB_INCLUDE_DIR)
 MARK_AS_ADVANCED(CLEAR ZLIB_LIBRARY)
+
 #
 # Find BZip2
 #
@@ -478,34 +479,11 @@ MARK_AS_ADVANCED(CLEAR BZIP2_LIBRARIES)
 # Find LZMA
 #
 IF(ENABLE_LZMA)
-  FIND_PACKAGE(LibLZMA)
-ELSE()
-  SET(LIBLZMA_FOUND FALSE) # Override cached value
-ENDIF()
-
-IF(LIBLZMA_FOUND)
   SET(HAVE_LIBLZMA 1)
   SET(HAVE_LZMA_H 1)
-  CMAKE_PUSH_CHECK_STATE()
-  SET(CMAKE_REQUIRED_INCLUDES ${LIBLZMA_INCLUDE_DIR})
-  SET(CMAKE_REQUIRED_LIBRARIES ${LIBLZMA_LIBRARIES})
-  INCLUDE_DIRECTORIES(${LIBLZMA_INCLUDE_DIRS})
-  LIST(APPEND ADDITIONAL_LIBS ${LIBLZMA_LIBRARIES})
-  # Test if a macro is needed for the library.
-  TRY_MACRO_FOR_LIBRARY(
-    "${LIBLZMA_INCLUDE_DIRS}" "${LIBLZMA_LIBRARIES}"
-    COMPILES
-    "#include <lzma.h>\nint main() {return (int)lzma_version_number(); }"
-    "WITHOUT_LZMA_API_STATIC;LZMA_API_STATIC")
-  IF(NOT WITHOUT_LZMA_API_STATIC AND LZMA_API_STATIC)
-    ADD_DEFINITIONS(-DLZMA_API_STATIC)
-  ENDIF(NOT WITHOUT_LZMA_API_STATIC AND LZMA_API_STATIC)
-  CMAKE_POP_CHECK_STATE()
-ELSE(LIBLZMA_FOUND)
-# LZMA not found and will not be used.
-ENDIF(LIBLZMA_FOUND)
-MARK_AS_ADVANCED(CLEAR LIBLZMA_INCLUDE_DIR)
-MARK_AS_ADVANCED(CLEAR LIBLZMA_LIBRARY)
+  LIST(APPEND ADDITIONAL_LIBS liblzma)
+  ADD_DEFINITIONS(-DLZMA_API_STATIC)
+ENDIF()
 
 #
 # Find LZO2
diff -rupN orig/libarchive/CMakeLists.txt patched/libarchive/CMakeLists.txt
--- orig/libarchive/CMakeLists.txt	2020-01-28 00:19:39.000000000 +0100
+++ patched/libarchive/CMakeLists.txt	2025-02-24 16:49:53.345455543 +0100
@@ -166,51 +166,6 @@ SET(libarchive_SOURCES
   xxhash.c
 )
 
-# Man pages
-SET(libarchive_MANS
-  archive_entry.3
-  archive_entry_acl.3
-  archive_entry_linkify.3
-  archive_entry_misc.3
-  archive_entry_paths.3
-  archive_entry_perms.3
-  archive_entry_stat.3
-  archive_entry_time.3
-  archive_read.3
-  archive_read_add_passphrase.3
-  archive_read_data.3
-  archive_read_disk.3
-  archive_read_extract.3
-  archive_read_filter.3
-  archive_read_format.3
-  archive_read_free.3
-  archive_read_header.3
-  archive_read_new.3
-  archive_read_open.3
-  archive_read_set_options.3
-  archive_util.3
-  archive_write.3
-  archive_write_blocksize.3
-  archive_write_data.3
-  archive_write_disk.3
-  archive_write_filter.3
-  archive_write_finish_entry.3
-  archive_write_format.3
-  archive_write_free.3
-  archive_write_header.3
-  archive_write_new.3
-  archive_write_open.3
-  archive_write_set_options.3
-  archive_write_set_passphrase.3
-  cpio.5
-  libarchive.3
-  libarchive_changes.3
-  libarchive_internals.3
-  libarchive-formats.5
-  mtree.5
-  tar.5
-)
-
 IF(WIN32 AND NOT CYGWIN)
   LIST(APPEND libarchive_SOURCES archive_entry_copy_bhfi.c)
   LIST(APPEND libarchive_SOURCES archive_read_disk_windows.c)
@@ -235,14 +190,9 @@ ELSEIF(ARCHIVE_ACL_SUNOS)
   LIST(APPEND libarchive_SOURCES archive_disk_acl_sunos.c)
 ENDIF()
 
-# Libarchive is a shared library
-ADD_LIBRARY(archive SHARED ${libarchive_SOURCES} ${include_HEADERS})
-TARGET_INCLUDE_DIRECTORIES(archive PUBLIC .)
-TARGET_LINK_LIBRARIES(archive ${ADDITIONAL_LIBS})
-SET_TARGET_PROPERTIES(archive PROPERTIES SOVERSION ${SOVERSION})
-
 # archive_static is a static library
 ADD_LIBRARY(archive_static STATIC ${libarchive_SOURCES} ${include_HEADERS})
+TARGET_INCLUDE_DIRECTORIES(archive_static PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 TARGET_LINK_LIBRARIES(archive_static ${ADDITIONAL_LIBS})
 SET_TARGET_PROPERTIES(archive_static PROPERTIES COMPILE_DEFINITIONS
   LIBARCHIVE_STATIC)
@@ -253,12 +203,9 @@ ENDIF(NOT WIN32 OR CYGWIN)
 
 IF(ENABLE_INSTALL)
   # How to install the libraries
-  INSTALL(TARGETS archive archive_static
+  INSTALL(TARGETS archive_static
           RUNTIME DESTINATION bin
           LIBRARY DESTINATION lib
           ARCHIVE DESTINATION lib)
-  INSTALL_MAN(${libarchive_MANS})
   INSTALL(FILES ${include_HEADERS} DESTINATION include)
 ENDIF()
-
-add_subdirectory(test)
diff -rupN orig/libarchive/archive_read_support_format_rar5.c patched/libarchive/archive_read_support_format_rar5.c
--- orig/libarchive/archive_read_support_format_rar5.c	2020-02-11 00:30:01.000000000 +0100
+++ patched/libarchive/archive_read_support_format_rar5.c	2025-02-24 14:29:00.203562989 +0100
@@ -3084,12 +3084,17 @@ static int do_uncompress_block(struct ar
 
 			continue;
 		}
-
+#ifndef _MSC_VER
+/*
+ * This is unreachable code (seemingly by design), and compilation on MSVC fails because of that,
+ * so we remove it when compiling with MSVC.
+ */
 		/* The program counter shouldn't reach here. */
 		archive_set_error(&a->archive, ARCHIVE_ERRNO_FILE_FORMAT,
 		    "Unsupported block code: 0x%x", num);
 
 		return ARCHIVE_FATAL;
+#endif
 	}
 
 	return ARCHIVE_OK;
