diff --git a/src/session.c b/src/session.c
index d4c1691d..c365290c 100644
--- a/src/session.c
+++ b/src/session.c
@@ -266,7 +266,10 @@ static int session_banner_send(LIBSSH2_SESSION *session)
 static int session_nonblock(libssh2_socket_t sockfd,   /* operate on this */
                             int nonblock /* TRUE or FALSE */ )
 {
-#ifdef HAVE_O_NONBLOCK  /* most recent unix versions */
+#if defined(_WIN32)
+    unsigned long flags = nonblock;
+    return ioctlsocket(sockfd, FIONBIO, &flags);
+#elif defined(HAVE_O_NONBLOCK)  /* most recent unix versions */
     int flags = fcntl(sockfd, F_GETFL, 0);
     if(nonblock)
         return fcntl(sockfd, F_SETFL, flags | O_NONBLOCK);
@@ -280,9 +283,6 @@ static int session_nonblock(libssh2_socket_t sockfd,   /* operate on this */
 #elif defined(HAVE_SO_NONBLOCK)  /* BeOS */
     long b = nonblock ? 1 : 0;
     return setsockopt(sockfd, SOL_SOCKET, SO_NONBLOCK, &b, sizeof(b));
-#elif defined(_WIN32)
-    unsigned long flags = nonblock;
-    return ioctlsocket(sockfd, FIONBIO, &flags);
 #else
     (void)sockfd;
     (void)nonblock;
